### ISSUE TITLE
missing TypeScript declaration file

### ISSUE THREAD

**Issue by Zamiell** [eslint-community/eslint-plugin-eslint-comments#214]
Now that ESLint 9 is released and everyone is migrating to flat config, importing the plugin into a flat config by following the official documentation actually results in a red squiggly line, because no declaration file exists (when using `checkJs` and/or `@ts-check`).

I suspect many/most users will be using TypeScript to validate that their new flat configs do not have typos in them. Thus, a declaration file would really be appreciated.

**Comment by jakec-dev** [comment]
Agreed on this, however in the meanwhile you can create a basic declaration file as below to avoid those red lines:

```
declare module '@eslint-community/eslint-plugin-eslint-comments/configs' {
  import type { Linter } from 'eslint';

  const recommended: Linter.Config;
  export = { recommended };
}
```

**Comment by silverwind** [comment]
My current workaround is:

```ts
declare module '@eslint-community/eslint-plugin-eslint-comments' {
  import type {Eslint} from 'eslint';
  const plugin: Eslint.Plugin;
  export = plugin;
}
```
